src/lib/gadgets/foreign-field.ts

```diff
@@ -79,8 +94,8 @@ function singleAdd(x: Field3, y: Field3, sign: Sign, f: bigint) {
     // do the add with carry
     // note: this "just works" with negative r01
     let r01 = collapse2(x_) + sign * collapse2(y_) - overflow * collapse2(f_);
-    let carry = r01 >> twoL;
-    r01 &= twoLMask;
+    let carry = r01 >> l2;
```

Comment:
maybe `combine` is a better name that `collapse` ?

Reply:
How about join()? That matches with split()

Reply:
went with combine